src/openapi-generator.ts

```diff
@@ -655,19 +655,52 @@ export class OpenAPIGenerator {
     zodSchema: ZodObject<ZodRawShape>,
     isNullable: boolean
   ): SchemaObject {
+    const extendedFrom = zodSchema._def.openapi?.extendedFrom;
+
     const propTypes = zodSchema._def.shape();
     const unknownKeysOption = zodSchema._unknownKeys as UnknownKeysParam;
 
     const requiredProperties = Object.entries(propTypes)
       .filter(([_key, type]) => !this.isOptionalSchema(type))
       .map(([key, _type]) => key);
 
+    const properties = mapValues(propTypes, propSchema =>
+      this.generateInnerSchema(propSchema)
+    );
+
+    if (extendedFrom) {
+      const alreadyRegistered = Object.keys(
+        this.schemaRefs[extendedFrom]?.properties ?? {}
+      );
+      const alreadyRequired = this.schemaRefs[extendedFrom]?.required ?? [];
+
+      const additionalProperties = omit(properties, alreadyRegistered);
+      const additionallyRequired = requiredProperties.filter(
+        prop => !alreadyRequired.includes(prop)
+      );
+
+      const additionalData = {
+        type: 'object' as const,
+
+        properties: additionalProperties,
+
+        required:
+          additionallyRequired.length > 0 ? additionallyRequired : undefined,
+      };
```

Comment:
We're skipping the `additionalProperties` check here. Can we somehow merge the two branches actually? e.g. if (extendedFrom), then just set alreadyRegistered and alreadyRequired and continue with the same logic regardless, just default those to empty, then wrap in `allOf` at the end if necessary?

Reply:
You are correct about `additionalProperties` I think. But for `nullable` it is more complicated from what I noticed:
```yaml
allOf:
  - $ref: <some-ref>
  - type: object
     properties:
        a:
          type: string
   nullable: true # This right here is wrong/does not produce any result
```
The above is based on the [docs](https://swagger.io/docs/specification/using-ref/)
![image](https://user-images.githubusercontent.com/44395908/189305617-386058a4-01bb-4c78-868e-50a1eb32b0d5.png)
I believe that if you are adding a nullable the result should be something like:
```yaml
anyOf:
  - type: 'null' # I am still unable to confirm that this is a valid syntax
  - allOf: 
     - $ref: <some-ref>
     - type: object
       properties:
          a:
            type: string
```

So basically I don't think we can join them as they need to be 2 separate flows